pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -83,43 +83,85 @@ func TestMirrors(t *testing.T) {
 	assert.True(t, reg.Mirrors[1].Insecure)
 }
 
-func TestRefMatchesPrefix(t *testing.T) {
+func TestRefMatchingSubdomainPrefix(t *testing.T) {
 	for _, c := range []struct {
 		ref, prefix string
-		expected    bool
+		expected    int
+	}{
+		// Check for subdomain matches
+		{"docker.io", "*.io", len("docker.io")},
+		{"docker.io/foo", "*.com", -1},
+		{"example.com/foo", "*.co", -1},
```

Comment:
Can you add one test for "example.com/foo" and "*.example.com"?

Reply:
done